src/v/archival/ntp_archiver_service.h

```diff
@@ -233,6 +256,22 @@ class ntp_archiver {
     /// STM manifest.
     ss::future<> apply_spillover();
 
+    // Request a flush operation of all current local data to cloud storage.
+    // This function can be used in combination with wait() to block until the
+    // flush operation is complete.
+    // Calling this function does not guarantee a flush will begin, in the case
+    // the partition is not the leader or is a read replica. It also does not
+    // guarantee that the flush will complete, in the case of a leadership
+    // change occurs during the operation. It is important to issue a call to
+    // wait() to be certain of the state post a call to flush().
+    flush_result flush();
```

Comment:
are there concurrency restrictions that need to be described here? the reason i ask is because below the wording of "current flush operation" is used.

Reply:
Expanded comment here to describe possible undesired behavior if `flush()` is called multiple times without user resolving `wait()` futures, as waiters will only be notified when the most recently issued flush has completed.